### PR TITLE
Clarify that release dates are approximate

### PR DESCRIPTION
**Pull request by jbw976**
As a follow-up here in the docs from https://github.com/crossplane/crossplane/pull/4211, this PR clarifies that the release dates are approximate and provides reasoning why.

### CODE REVIEW

**Review by jeanduplessis — CHANGES_REQUESTED** [review]
@jbw976 would you mind having a look at the Vale errors for this new section

```
 86:4   warning  'Release Dates' should use      Google.Headings        
                 sentence-style capitalization.                         
 88:12  warning  'is released' may be passive    write-good.Passive     
                 voice. Use active voice if you                         
                 can.                                                   
 88:29  warning  'quarterly' is a weasel word!   write-good.Weasel      
 89:1   warning  Avoid using future tense:       gitlab.FutureTense     
                 "will happen". Use present                             
                 tense instead.                                         
 90:43  warning  Remove this word from the       gitlab.Wordy           
                 sentence: "Please".                                    
 91:26  warning  'A number of' is too wordy.     write-good.TooWordy   
 ```

**Comment by jbw976** [comment]
@jeanduplessis I fixed all [Vale errors](https://github.com/crossplane/docs/actions/runs/5463253116/jobs/9943703502?pr=481) within the entire doc, and fixed all warnings within this new section:

```
✖ 0 errors, 55 warnings and 0 suggestions in 1 file.
```

Can you take another look now sir? 🙇

**Review by jeanduplessis — APPROVED** [review]
Thanks @jbw976 LGTM 🚀
